plugins/modules/digital_ocean_domain.py

```diff
@@ -155,7 +161,11 @@ def destroy_domain(self):
             return json
 
     def edit_domain_record(self, record):
-        params = {"name": "@", "data": self.module.params.get("ip")}
+        if self.module.params.get("ip"):
+            params = {"name": "@", "data": self.module.params.get("ip")}
+        if self.module.params.get("ip6"):
+            params = {"name": "@", "data": self.module.params.get("ip6")}
```

Comment:
The `params` here is being overwritten; does this mean that this module only supports setting one type of apex record at a time?

Reply:
The [API docs](https://docs.digitalocean.com/reference/api/api-reference/#operation/create_domain_record) aren't overly clear, but yes, it looks like the `type` parameter can be *one of* `['A', 'AAAA', 'CNAME', ...]`.

Reply:
The [API docs](https://docs.digitalocean.com/reference/api/api-reference/#operation/create_domain_record) aren't overly clear, but yes, it looks like the `type` parameter can be *one of* `['A', 'AAAA', 'CNAME', ...]`.

Reply:
So should we make this an `elif`, then, and document the idea that `ip` will be preferred over `ip6`? (as written, if both are supplied then only `ip6` will be updated)

Reply:
I don't think it matters; this module uses `mutually_exclusive=[("ip", "ip6")],` -- the module will fail if both arguments are specified.